src/data/move.ts

```diff
@@ -5123,6 +5122,43 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
   }
 }
 
+/**
+ * Attr used by parting shot, it is a combo of ForceSwitchOut and StatChange with a special getCondition()
+ */
+export class PartingShotAttr extends ForceSwitchOutAttr {
+  private statChange = new StatStageChangeAttr([ Stat.ATK, Stat.SPATK ], -1, false, null, true, true, MoveEffectTrigger.PRE_APPLY);
+  private canLowerStats = true;
+
+  // using inherited constructor
+
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): Promise<boolean> {
+    // apply stat change, conditionally apply switch-out
+    this.statChange.apply(user, target, move, args);
+    if (this.canLowerStats) {
+      return super.apply(user, target, move, args);
+    } else {
+      return new Promise(resolve => {
+        resolve(false);
+      });
+    }
+  }
+
+  getCondition(): MoveConditionFunc {
+    return (user, target, move) => {
+      // conditions on if move should fail or not don't depend on if user is able to switch
+      // getCondition() is called before move is applied: move will only switch out if canLowerStats === true
+      if (target.hasAbility(Abilities.CLEAR_BODY, true, false) ||
```

Comment:
```suggestion
      if (target.hasAbilityWithAttr(ProtectStatAbAttr) ||
```
The existing condition doesn't include White Smoke, while this covers both abilities

Reply:
thanks, added a test for that too